misc/openapi-ballerina/modules/openapi-to-ballerina-generator/src/main/java/org/ballerinalang/openapi/model/BallerinaOpenApi.java

```diff
@@ -127,8 +128,37 @@ private void setPaths(OpenAPI openAPI) throws BallerinaOpenApiException {
                     }
                 });
             }
-            paths.add(new AbstractMap.SimpleEntry<>(path.getKey(), balPath));
+            String resourcePath;
+            // Use ballerina string templating if there are path params.
+            if (hasPathParams(path.getValue())) {
+                resourcePath = "string `" + path.getKey().replace("{", "${") + "`";
+            } else {
+                resourcePath = "\"" + path.getKey() + "\"";
+            }
+            paths.add(new AbstractMap.SimpleEntry<>(resourcePath, balPath));
+        }
+    }
+
+    /**
+     * Checks if there are any path parameters in the operations.
+     * @param path  The path item in the OpenAPI spec.
+     * @return True if there are path parameters, else false.
+     */
+    private boolean hasPathParams(PathItem path) {
+        if (path.getParameters() != null && path.getParameters().size() > 0) {
+            return path.getParameters().stream().anyMatch(parameter -> parameter.getIn() != null && parameter.getIn()
+                    .equals(GeneratorConstants.PATH));
+        }
+        if (path.readOperations().size() > 0) {
+            return path.readOperations().stream().anyMatch(operation -> {
+                if (operation.getParameters() != null && operation.getParameters().size() > 0) {
+                    return operation.getParameters().stream().anyMatch(parameter -> parameter.getIn() != null &&
+                            parameter.getIn().equals(GeneratorConstants.PATH));
+                }
+                return false;
+            });
         }
+        return false;
```

Comment:
IMO better to do an early return in here. WDYT? 
```suggestion
        if (path.readOperations().isEmpty()) {
            return false;
        }
        return path.readOperations().stream().anyMatch(operation -> {
            if (operation.getParameters() != null && operation.getParameters().size() > 0) {
                return operation.getParameters().stream().anyMatch(parameter -> parameter.getIn() != null &&
                        parameter.getIn().equals(GeneratorConstants.PATH));
            }
            return false;
        });
    }
```

Reply:
Also, don't we need to validate whether `path.readOperations()` is null ?

Reply:
yeah we don't need to validate with null :) it return list